### ISSUE TITLE
sig missing from handshake's auth spec

### ISSUE THREAD

**Issue by ryandotsmith** [ethereum/devp2p#218]
```
auth = auth-size || enc-auth-body
auth-size = size of enc-auth-body, encoded as a big-endian 16-bit integer
auth-vsn = 4
auth-body = [sig, initiator-pubk, initiator-nonce, auth-vsn, ...]
enc-auth-body = ecies.encrypt(recipient-pubk, auth-body || auth-padding, auth-size)
auth-padding = arbitrary data
```

Looking at various implementations, it seems like sig is: `shared-secret ^ initiator-nonce` and shared-secret is: `ecdh.agree(privkey, remote-pubk)`

**Comment by ryandotsmith** [comment]
Also, there might need to be a definition for `auth-padding` and clarification on `auth-size`.

**Comment by ryandotsmith** [comment]
Ah, I suppose that auth-size can be known prior to ecies encryption using: `len(auth-body || auth-padding) + 113`. Does that sound correct?

**Comment by kdeme** [comment]
> Ah, I suppose that auth-size can be known prior to ecies encryption using: `len(auth-body || auth-padding) + 113`. Does that sound correct?

Yes, 113 is the fixed ecies overhead which can be used to pre-calculate the total size.
